### ISSUE TITLE
[TODO] Clean up `controlPanels/sections`  and `controls` in superset-frontend

### ISSUE THREAD

**Issue by ktmud** [apache/superset#13419]
Part of these files are moved to `@superset-ui/chart-controls` but still incomplete.

- [superset-frontend/src/explore/controlPanels/sections.jsx](https://github.com/apache/superset/blob/2a587f6cb1e598ef2d90ff3e60e97094ceec2ca1/superset-frontend/src/explore/controlPanels/sections.jsx) -> [packages/superset-ui-chart-controls/src/sections.tsx](https://github.com/apache-superset/superset-ui/blob/59283de7b0c9110c5dea64ac0ff4ef8d4554456d/packages/superset-ui-chart-controls/src/sections.tsx)
- [superset-frontend/src/explore/controls.jsx](https://github.com/apache/superset/blob/3c62069bbb0d07ff0194fbb159c2e04fd98fafc2/superset-frontend/src/explore/controls.jsx#L281) -> [packages/superset-ui-chart-controls/src/shared-controls/index.tsx](https://github.com/apache-superset/superset-ui/blob/bd4d592299ff31c9f036feb93f93b529eb66fb8e/packages/superset-ui-chart-controls/src/shared-controls/index.tsx)

Diverging changes in these almost identical files have caused [confusion](https://github.com/apache/superset/discussions/13404) and bugs. We should clean up the version in `superset-frontend` and port any [subsequent changes](https://github.com/apache/superset/commits/master/superset-frontend/src/explore/controlPanels/sections.jsx) after the file was initially copied over to `sueprst-ui/chart-controls`.

**Comment by Red-0111** [comment]
Hi, I would like to work on this issue, please tell me how to proceed :)

**Comment by zuzana-vej** [comment]
It's open for over a year; is this still relevant / anyone plans to work on it? @ktmud @Red-0111 ?

**Comment by ktmud** [comment]
Yes, we still need to do this. It should be easier to do now that everything is in one monorepo.

**Comment by rusackas** [comment]
@yousoph @kgabryje @michael-s-molina @eric-briscoe this is indeed a nice little piece of tech debt that still needs to be solved. Is it on anyone's radar at the moment?

**Comment by rusackas** [comment]
This hairball still exists. It's not a _bug_ exactly so I'm tempted to close this (especially since it's also stale) but it's a reminder that this work still needs to be done. A LOT of work still needs to be done to clean up control panels, in fact.

**Comment by nvmnghia** [comment]
@rusackas What needs to be done? I'm interested in working on this.

**Comment by rusackas** [comment]
> @rusackas What needs to be done? I'm interested in working on this.

Pretty much what the issue describes... the two files linked should be merged into the monorepo packages, rather than being a part of Superset's main codebase. The dependencies here might be fairly intricate (TBD) so I worry this will be a delicate surgery. But then again, there's probably no need to move the _entire_ file... even moving one piece of code to the right spot is a step in the right direction.

**Comment by nvmnghia** [comment]
The current files:

- [superset-frontend/src/explore/controlPanels/sections.tsx][1] -> [superset-frontend/packages/superset-ui-chart-controls/src/sections/sections.tsx][2]

After glancing these 2 files, I think there's nothing more to refactor. The only duplicated bit is this:

```tsx
export const colorScheme: ControlPanelSectionConfig = {
  label: t('Color scheme'),
  controlSetRows: [['color_scheme']],
};
```

The rest of those are different enough to justify the duplicated code.

  [1]: https://github.com/apache/superset/blob/1d571ec9e6e769d9715ee1e5c398aca079ba6207/superset-frontend/src/explore/controlPanels/sections.tsx
  [2]: https://github.com/apache/superset/blob/1d571ec9e6e769d9715ee1e5c398aca079ba6207/superset-frontend/packages/superset-ui-chart-controls/src/sections/sections.tsx

**Comment by rusackas** [comment]
Right... I haven't looked closely at this in forever, so I don't know all the details/difficulties of this effort, but the general idea is that ALL code related to control panels should be moved to the package(s) for maximum reusability and minimum code duplication.

**Comment by rusackas** [comment]
This issue, and all effort behind it AFAIK, seems to have gone stale. I'll go ahead and close this out. I think it's fair to say that the control panels still need a TON of design and engineering work to make them more consistent, more sensible, more DRY, and more... modern/React-y. This seems like the job for a SIP and a whole project kanban, so I'll go ahead and close this out until there's some more significant time to dedicate to the effort.
